readers/common/src/main/java/com/criteo/hadoop/garmadon/protobuf/ProtoConcatenator.java

```diff
@@ -0,0 +1,152 @@
+package com.criteo.hadoop.garmadon.protobuf;
+
+import com.github.os72.protobuf.dynamic.DynamicSchema;
+import com.github.os72.protobuf.dynamic.MessageDefinition;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+public class ProtoConcatenator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProtoConcatenator.class);
+
+    /**
+     * Concatenate Protobuf messages into a single Protobuf message.
+     * /!\ Doesn't handle embedded objects and repeated fields /!\
```

Comment:
looks like that you manage it, no?

Reply:
I always underestimate my code :smile: Fixed